mesonbuild/utils/core.py

```diff
@@ -47,6 +47,15 @@ def __init__(self, *args: object, file: T.Optional[str] = None,
         self.lineno = lineno
         self.colno = colno
 
+    def set_from_node(self, node: BaseNode) -> None:
```

Comment:
Would update_node() be a better name?

Reply:
Maybe `update_position`? I guess the fact that update from a `BaseNode` is kinda an artifictial limitation. There's not reason that you couldn't use a Protocol here instead